src/analysis/normed/field/basic.lean

```diff
@@ -1021,3 +1023,25 @@ lemma norm_one_class.induced {F : Type*} (R S : Type*) [ring R] [semi_normed_rin
 { norm_one := (congr_arg norm (map_one f)).trans norm_one }
 
 end induced
+
+namespace subring_class
+
+variables {S R : Type*} [set_like S R]
+
+instance to_semi_normed_ring [semi_normed_ring R] [subring_class S R] (s : S) :
+  semi_normed_ring s :=
+semi_normed_ring.induced s R (subring_class.subtype s)
```

Comment:
Do we have the normed_group versions already?

Reply:
> We have the version for `subgroup`s, but not `subgroup_class`. We don't currently have the ring versions even for `subring`, and I am only adding them for `subring_class`.